scipy/stats/_binned_statistic.py

```diff
@@ -650,6 +643,27 @@ def binned_statistic_dd(sample, values, statistic='mean',
     return BinnedStatisticddResult(result, edges, binnumbers)
 
 
+def _calc_binned_statistic(Vdim, binnumbers, result, values, stat_func):
+    unique_bin_numbers = np.unique(binnumbers)
+    for vv in builtins.range(Vdim):
+        bin_map = _create_bin_map(binnumbers, unique_bin_numbers, values, vv)
+        for i in unique_bin_numbers:
+            result[vv, i] = stat_func(bin_map[i])
```

Comment:
I think there is but asking so that you can confirm, is there ***always*** a `callable()` check already done in outer scope?

Reply:
Do you mean L606-L613 check in the latest code? Yes, it already done when `callable` passed as a statistic.

Reply:
I see it as line 520 in master currently, 

https://github.com/scipy/scipy/blob/master/scipy/stats/_binned_statistic.py#L520

I think you are referring to the block of the if/elif/.../elif where it falls to the `callable` after checking keywords.